### PR TITLE
DELETE /post/:id  Fixes #28

### LINKED ISSUE #28: DELETE	/post/:id
This endpoint should delete the requested post.
The user should be able to delete their own posts only.

Add documentation and testing.

### PR DESCRIPTION
**Pull request by khaldarov**
- Fixes #28.
- Includes ~~tests and~~ documentation for #28.
- Organizes `.yaml` files.
- Fixes some typos.
- Fixes login and profile documentations.

### CODE REVIEW

**Comment by khaldarov** [comment]
## It'd be easier for reviewers to only review the first commit: it includes the required changes. The others were optional fixes I made.

File: src/controllers/product.js [hunk 1]
```diff
@@ -40,6 +41,38 @@ const addNewProduct = async (req, res) => {
   }
 };
 
+const deleteProduct = async (req, res) => {
+  try {
+    const singlePost = await ProductModel.findById(req.params.id);
+
+    if (singlePost) {
```
Comment (khatibAmjad) [inline src/controllers/product.js]: if we agreed not to delete the received product I think we should check for the isTransactionCompleted field too.

File: src/controllers/product.js [hunk 2]
```diff
@@ -40,6 +41,38 @@ const addNewProduct = async (req, res) => {
   }
 };
 
+const deleteProduct = async (req, res) => {
+  try {
+    const singlePost = await ProductModel.findById(req.params.id);
```
Comment (khatibAmjad) [inline src/controllers/product.js]: I think it'd be better if we stick to the product instead of post.

File: src/controllers/product.js [hunk 3]
```diff
@@ -40,6 +41,38 @@ const addNewProduct = async (req, res) => {
...
+          });
+        }
+        await singlePost.deleteOne();
+        return res
+          .status(200)
+          .json({ message: 'Product deleted successfully' });
+      }
+      return res
+        .status(401)
+        .json({ message: 'You are not authorized to perform this actions' });
```
Comment (khatibAmjad) [inline src/controllers/product.js]: action*

File: src/docs/authentication.yaml [hunk 4]
```diff
@@ -91,6 +91,41 @@ paths:
           description: 'Bad request (either validation or database error)'
         '500':
           description: 'Internal server error'
+  /api/auth/login:
```
Comment (khatibAmjad) [inline src/docs/authentication.yaml]: I already changed this and merged it to the dev branch.

File: src/routes/product.js [hunk 5]
```diff
@@ -5,5 +5,6 @@ const productController = require('../controllers/product');
 const router = express.Router();
 
 router.post('/', productMiddleware, productController.addNewProduct);
+router.delete('/:id', productController.deleteProduct);
```
Comment (khatibAmjad) [inline src/routes/product.js]: Can we change the `:id` to `:productId` so the consumer doesn't enter the userId

File: src/docs/product.yaml [hunk 6]
```diff
@@ -64,6 +64,31 @@ paths:
           description: 'Bad request (either validation or database error)'
         '500':
           description: 'Internal server error'
+  /api/product/{id}:
+    delete:
+      tags:
+        - CRUD products
```
Comment (khatibAmjad) [inline src/docs/product.yaml]: Our new tag for this route is Product

**Review by khatibAmjad — DISMISSED** [review]

File: src/docs/product.yaml [hunk 7]
```diff
@@ -64,6 +64,31 @@ paths:
           description: 'Bad request (either validation or database error)'
         '500':
           description: 'Internal server error'
+  /api/product/{id}:
```
Comment (okurezgi) [inline src/docs/product.yaml]: i think we should change the id part to productId (i did that in patch)

File: src/docs/product.yaml [hunk 8]
```diff
@@ -64,6 +64,31 @@ paths:
...
+    delete:
+      tags:
+        - CRUD products
+      description: 'Deletes a product'
+      parameters:
+        - name: 'id'
```
Comment (okurezgi) [inline src/docs/product.yaml]: also productId would be better

**Review by okurezgi — CHANGES_REQUESTED** [review]

File: src/docs/authentication.yaml (hunk 4 above)
Comment (khaldarov) [reply src/docs/authentication.yaml]: No problem. It had flaws. It'll be fixed with this PR.

**Review by okurezgi — APPROVED** [review]
Adnan great job and thanks for being so fast

**Review by halitbatur — APPROVED** [review]

**Pull request merged**